src/react/utils.js

```diff
@@ -197,6 +204,20 @@ export function convertSimpleClassComponentToFunctionalComponent(
   invariant(prototype.descriptor);
   prototype.descriptor.configurable = true;
   Properties.DeletePropertyOrThrow(realm, componentType, "prototype");
+
+  // fix the length as we've changed the arguments
+  let lengthProperty = GetDescriptorForProperty(componentType, "length");
+  invariant(lengthProperty);
+  lengthProperty.writable = false;
+  lengthProperty.enumerable = false;
+  lengthProperty.configurable = true;
+  // ensure the length value is set to the new value
+  let lengthValue = Get(realm, componentType, "length");
+  invariant(lengthValue instanceof NumberValue);
+  lengthValue.value = 2;
```

Comment:
Why does this matter?

Reply:
Because of `val.hasDefaultLength()` checks the value against the arguments to validate if it doesn't need to be serialized out.

https://github.com/trueadm/prepack/blob/fixes-class-component-replacement/src/serializer/ResidualHeapInspector.js#L89

Reply:
But that should still work, so does the unit test in this PR work before and after this change? If so, then we should add another test that tests this specifically (e.g. by reading the function length).

Reply:
The new test fails before. The reason is, we change the arguments of the function.

Reply:
I just ran it locally and it doesn't fail even without the fix. What am I missing?

Reply:
Sorry, I was a n00b and forgot to `git add` the test that fails. :D Done that now!

Reply:
Can you explain why this matters to the serializer? I'm worried that this won't work if the length is explicitly defined, e.g. by some compiler output. I'm also just worried that I don't understand it. :)

Reply:
It needs to be 2 because I explicitly set the arguments array to be two elements (it's fixed to always be this intentionally):

https://github.com/trueadm/prepack/blob/fixes-class-component-replacement/src/react/utils.js#L224

If I do not set it, and the arguments length is different, it means the serialization output of this functional class component will include 3-4 extra lines of AST to use `defineProperty` to set it.

Reply:
Yea I understand that we'll want this change regardless for perf/size reasons but I wonder why it breaks right now. Might be revealing some other bug.

Reply:
@trueadm Explained to me that this is actually unrelated to the fix. It is the `.kind` change that is the fix to the bug and this line is just an optimization.